### DISCUSSION TITLE
Create instance of `WindowEvent::KeyboardInput`.

### DISCUSSION THREAD

**Discussion by TornaxO7** [rust-windowing/winit#3176] [Q&A] [answered]
Hello! [I'm currently trying to bump up the `winit` dependency in wgpu](https://github.com/gfx-rs/wgpu/pull/4202).

As you can see here, we are [emulating a WindowEvent](https://github.com/gfx-rs/wgpu/blob/trunk/examples/common/src/framework.rs#L571-L580):
```rust
        example.update(winit::event::WindowEvent::KeyboardInput {
                            input: winit::event::KeyboardInput {
                                scancode: 0,
                                state: winit::event::ElementState::Pressed,
                                virtual_keycode: Some(winit::event::VirtualKeyCode::Space),
                                modifiers: winit::event::ModifiersState::empty(),
                            },
                            device_id: unsafe { winit::event::DeviceId::dummy() },
                            is_synthetic: false,
                        });
```
I tried to update this part of code and got the following result:
```rust
                        example.update(winit::event::WindowEvent::KeyboardInput {
                            event: KeyEvent {
                                physical_key: PhysicalKey::Code(KeyCode::Space),
                                state: ElementState::Pressed,
                                logical_key: Key::Named(NamedKey::Space),
                                text: None,
                                location: KeyLocation::Standard,
                                repeat: false,
                                platform_specific: false,
                            },
                            device_id: unsafe { winit::event::DeviceId::dummy() },
                            is_synthetic: false,
                        });
```
However, I've got a problem with `platdform_specific: false` here, since it's [a private attribute](https://docs.rs/winit/0.29.2/src/winit/event.rs.html#781-787) and I couldn't find a `new` constructor or something like that for it. May I ask what's the proper way to fix this with the latest `winit` update?

**Comment by kchibisov** [comment, marked as answer]
This is so cursed...

Why not just use a `logical_key`? Maybe some other state you actually need instead of doing synthetic event like that.

I could think of creating such events for testing purposes, and hide the platform_specific field for tests, but not sure what to do with general code, since platform specific fields sometimes are required...

I'd just suggest to not do what you're doing in this example, since there's clearly some abuse going on.

**Reply by TornaxO7** [reply]
well then. I'll try to refactor this as well (or try something else). Thank you for the information!
